nncf/common/tensor_statistics/aggregator.py

```diff
@@ -50,21 +49,15 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
 
         merged_statistics = self._get_merged_statistic_points(self.statistic_points, model, graph)
         transformation_layout = self._get_transformation_layout_extra_outputs(merged_statistics)
-        model_with_outputs = model_transformer.transform(transformation_layout)
+        model_with_outputs: ModelTransformer = model_transformer.transform(transformation_layout)
         engine = factory.EngineFactory.create(model_with_outputs)
 
-        dataset_length = self.dataset.get_length()
-        total = (
-            min(dataset_length or self.stat_subset_size, self.stat_subset_size)
-            if self.stat_subset_size is not None
-            else None
-        )
```

Comment:
There was logic before: if `self.stat_subset_size` is None then subset size is equal to the length of the whole dataset. As default value is 0 for `self.stat_subset_size` and this check is gone, I don't see the place where this logic is implemented. Could you please explain where is it?

Reply:
The logic to set `self.stat_subset_size` to the length of the whole dataset if it is None is not explicitly present. However, there are steps taken during the registration of statistic points that indirectly handle the calculation of `self.stat_subset_size`.

<img width="838" alt="Screenshot 2024-02-12 at 3 44 35 PM" src="https://github.com/openvinotoolkit/nncf/assets/80677673/20f3445c-e3dc-47a1-a87a-1c501fc16ee5">

In this part of the code, it iterates over the registered statistic points and their associated tensor collectors. For each tensor collector, it checks if `self.stat_subset_size` is None. If it is None, it sets `self.stat_subset_size` to the num_samples attribute of the tensor collector.
If `self.stat_subset_size` is not None, it compares it with the num_samples attribute of the tensor collector and updates `self.stat_subset_size` to be the maximum of the two values.
This logic effectively sets `self.stat_subset_size` based on the maximum number of samples encountered among all tensor collectors associated with the registered statistic points. While it doesn't directly set it to the length of the whole dataset if `self.stat_subset_size` is None, it achieves a similar effect by considering the maximum number of samples encountered during registration.

Reply:
Got you. But despite the fact that this should work in every case, I prefer explicit double check